analysis/wwz/run_wwz4l.py

```diff
@@ -324,60 +325,64 @@ def LoadJsonToSampleName(jsonFile, prefix):
             fileset[name]["metadata"] = {"dataset": name}
     print(fileset)
+    print("Number of datasets:",len(fdict))
 
-    with Client() as _:
-        print("Number of datasets:",len(fdict))
-        dataset_runnable, dataset_updated = preprocess(
-            fileset,
-            maybe_step_size=50_000,
-            align_clusters=False,
-            files_per_batch=1,
-            skip_bad_files=True,
-            #calculate_form=True,
-        )
-
-        print('\n\nHere after preprocess!!!')
-
-        # "runnable" means the dataset containing only files that were successfully opened, this either copies or creates metadata entries depending on if you defined some already
-        # moreover, you could save these two dicts to a file and just read them in later, only ever running preprocess once every time you update your analysis's datasets.
-        outputs, reports = apply_to_fileset(
-            processor_instance, 
-            fileset, 
-            uproot_options={"allow_read_errors_with_report": True}
-        )
-        print('\n\nHere after apply_to_fileset')
-
-        coutputs, creports = dask.compute(outputs, reports) # , scheduler=taskvine
-
-        print('\n\nHere after compute!!!')
 
-        exit()
+    #### Try with distributed Client ####
+    distributed_client = 0 # Tmp
+    if distributed_client:
+        with Client() as _:
 
-    ####################################3
+            # Run preprocess
+            print("\nRunning preprocess...")
+            dataset_runnable, dataset_updated = preprocess(
+                fileset,
+                maybe_step_size=50_000,
+                align_clusters=False,
+                files_per_batch=1,
+                skip_bad_files=True,
+                #calculate_form=True,
+            )
 
-    # Works
-    '''
-    # Create dict of events objects
-    print("Number of datasets:",len(fdict))
-    events_dict = {}
-    for name, fpaths in fdict.items():
-        events_dict[name] = NanoEventsFactory.from_root(
-            {fpath: "/Events" for fpath in fpaths},
-            schemaclass=NanoAODSchema,
-            metadata={"dataset": name},
-        ).events()
+            # Run apply_to_fileset
+            print("\nRunning apply_to_fileset...")
+            histos_to_compute, reports = apply_to_fileset(
+                processor_instance,
+                fileset,
+                uproot_options={"allow_read_errors_with_report": True}
+            )
 
+            print("\nRunning compute...")
+            output, report = dask.compute(histos_to_compute, reports) # , scheduler=taskvine
 
-    # Get and compute the histograms
-    histos_to_compute = {}
-    for json_name in fdict.keys():
-        print(f"Getting histos for {json_name}")
-        histos_to_compute[json_name] = processor_instance.process(events_dict[json_name])
+            print("\nColumns used...")
+            used_columns = dak.necessary_columns(histos_to_compute)
```

Comment:
In fact, `necessary_columns` is predictive. You can call it before you run any actual analysis to see what it is going to read.

Reply:
Ahh, that is good to know. Thanks!